include/cuda/std/detail/libcxx/include/optional

```diff
@@ -606,12 +657,14 @@ private:
     // LWG2756: conditionally explicit conversion from _Up
     struct _CheckOptionalArgsConstructor {
       template <class _Up>
+      _LIBCUDACXX_INLINE_VISIBILITY
       static constexpr bool __enable_implicit() {
```

Comment:
My spider senses are tingling that this will break a lot of older supported compilers hard. 

If I remember correctly, I had to revert all those changes and move to inline variables / structs

Reply:
Can you elaborate? I'm not seeing any failures around this in the CI or locally.

Reply:
This was just something that broke tuple maybe it was just with older 11.X CTKs so that should not be an issue here